File: push/src/main/java/com/xiaomi/xmsf/push/utils/ConfigValueConverter.java
package com.xiaomi.xmsf.push.utils;

import com.elvishew.xlog.Logger;
import com.elvishew.xlog.XLog;
import com.nihility.utils.Singleton;
import com.xiaomi.xmpush.thrift.XmPushActionContainer;
import com.xiaomi.xmsf.utils.ConvertUtils;

public class ConfigValueConverter {
    private static final Logger logger = XLog.tag(ConfigValueConverter.class.getSimpleName()).build();

    public static ConfigValueConverter getInstance() {
        return Singleton.instance();
    }

    public <T> Object convert(Object root, String[] path, T value) {
        if (path.length == 1 && "pushAction".equals(path[0])) {
            try {
                XmPushActionContainer container = (XmPushActionContainer) root;
                return ConvertUtils.getResponseMessageBodyFromContainer(container,
                        RegSecUtils.getRegSec(container));
            } catch (Throwable e) {
                logger.e("parse pushAction failed", e);
                return null;
            }
        }
        return value;
    }
}

File: push/src/main/java/com/xiaomi/xmsf/push/utils/ConfigValueConverter.kt
package com.xiaomi.xmsf.push.utils

import com.elvishew.xlog.Logger
import com.elvishew.xlog.XLog
import com.xiaomi.xmpush.thrift.XmPushActionContainer
import com.xiaomi.xmsf.utils.ConvertUtils

class ConfigValueConverter {
    fun <T> convert(root: Any?, path: Array<String>, value: T): Any? {
        if (path.size == 1 && "pushAction" == path[0]) {
            return try {
                val container = root as XmPushActionContainer
                ConvertUtils.getResponseMessageBodyFromContainer(container, RegSecUtils.getRegSec(container))
            } catch (e: Throwable) {
                logger.e("parse pushAction failed", e)
                null
            }
        }
        return value
    }

    companion object {
        private val logger: Logger = XLog.tag(ConfigValueConverter::class.java.simpleName).build()
    }
}
